lib/data/Util.js

```diff
@@ -111,9 +111,11 @@ export function arrTraverse(arr) {
 			default:
 				break;
 		}
-		result += '.';
+
+		if (j < arr.length - 1) {
+			result += '.';
+		}
```

Comment:
Checking `j' in every iteration seems to be inefficient.
How about leaving this line unchanged and modifying the original line 116 as the following?
```
    if (result.endsWith(".")) {
        result = result.slice(0, -1);
    }
```

Reply:
I considered something like this, but in the end I thought not adding the extra '.' at all was cleaner. Happy to go with this approach though, I'll push a change shortly